### PR TITLE
Specify dependencies in setup.py

### PR DESCRIPTION
**Pull request by ecrews**
# Changes
* _All_ package dependencies are now specified in `setup.py`. Now you can simply install by running `pip install -e .` (use `pip install -e .[dev]` to install dev dependencies.)
* Library metadata is no longer read from `voxel51/version.json` to avoid circular dependency of the install process on the library itself!

### CODE REVIEW

**Review by brimoor — COMMENTED** [review]
@ecrews thoughts on leaving `requirements.txt` in the repo as a fallback option for installing a frozen version of the dependencies that are known to work?

**Comment by ecrews** [comment]
> @ecrews thoughts on leaving `requirements.txt` in the repo as a fallback option for installing a frozen version of the dependencies that are known to work?

I'm fine with that.

File: setup.py [hunk 1]
```diff
@@ -8,22 +8,44 @@
 '''
 from setuptools import setup, find_packages
 
-import voxel51.constants as voxc
-
 
+# @note version info should be kept in-sync with `voxel51/version.json`
```
Comment (ecrews) [inline setup.py]: Why not just get this information from `voxel51/version.json` directly?

Comment (brimoor) [reply setup.py]: Just for consistency with how we're now doing it in other libraries.

Is it possible to access this `setup` information programmatically after the package is installed? In that case, I'd like to delete `version.json` and have this file be the source of truth for that information.

**Review by brimoor — APPROVED** [review]
LGTM

File: setup.py (hunk 1 above)
Comment (ecrews) [reply setup.py]: There's a library in Python 3.8 which apparently does this: https://docs.python.org/3.8/library/importlib.metadata.html, which has been backported here: https://importlib-metadata.readthedocs.io/en/latest/

**Review by brimoor — APPROVED** [review]
tested and LGTM

**Pull request merged**